packages/amplify-graphql-api-construct/src/amplify-graphql-definition.ts

```diff
@@ -88,11 +89,38 @@ export class AmplifyGraphqlDefinition {
     if (definitions.length === 1) {
       return definitions[0];
     }
+
+    // A strategy will be present multiple times in a given definition: once per model. We'll create a unique list per definition to ensure
+    // no reuse across definitions.
+    let combinedStrategyNames: string[] = [];
+    for (const definition of definitions) {
+      const definitionStrategyNames = new Set<string>();
+      for (const strategy of Object.values(definition.dataSourceStrategies)) {
+        if (!isSqlStrategy(strategy)) {
+          continue;
+        }
+        const strategyName = strategy.name;
+        if (combinedStrategyNames.includes(strategyName)) {
+          throw new Error(
+            `A SQL-based ModelDataSourceStrategy name ('${strategyName}') cannot be shared between definitions. To specify a SQL-based API with schemas across multiple files, use 'fromFilesAndStrategy'`,
+          );
+        }
+        definitionStrategyNames.add(strategyName);
```

Comment:
can we have models with DDB and SQL strategies within a definition?

Reply:
Types must be unique across all definitions, regardless of their data source binding.